axlearn/common/gradient_accumulation.py

```diff
@@ -0,0 +1,298 @@
+# Copyright © 2024 Apple Inc.
+"""This module provides functions to decorate a ForwardFn to allow for a minibatched
+version that enables gradient accumulation.
+"""
+import functools
+from typing import Any, Callable, Optional, Tuple
+
+import jax
+import numpy as np
+from jax import numpy as jnp
+
+from axlearn.common import utils
+from axlearn.common.config import ConfigOr, maybe_instantiate
+from axlearn.common.metrics import MetricAccumulator
+from axlearn.common.update_transformation import ForwardFn, ForwardOutputs
+from axlearn.common.utils import Nested, Tensor, input_partition_spec, with_sharding_constraint
+
+
+def _compute_minibatch_size(input_batch: Nested[Tensor], *, steps: int) -> int:
+    """Utility function to compute minibatch size from input batch.
+
+    Args:
+        input_batch: A ForwardFn input_batch.
+        steps: Integer steps to divide inputs with.
+
+    Returns:
+        An integer representing minibatch size.
+
+    Raises:
+        ValueError if the input batch is not divisible by steps.
```

Comment:
Sorry for missing this earlier, but we should document the other raise conditions too.

Reply:
sounds good, let me update it now

Reply:
done.